tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_hemv_kk.hpp

```diff
@@ -0,0 +1,290 @@
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+	 class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class Triangle,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+requires (Layout_A::template mapping<std::experimental::extents<numRows_A, numCols_A>>::is_always_unique())
+void hermitian_matrix_vector_product(kokkos_exec<ExeSpace> kexe,
+				      std::experimental::mdspan<
+					ElementType_A,
+					std::experimental::extents<numRows_A, numCols_A>,
+					Layout_A,
+				        std::experimental::default_accessor<ElementType_A>
+				      > A,
+				      Triangle tr,
+				      std::experimental::mdspan<ElementType_x,
+					std::experimental::extents<ext_x>,
+					Layout_x,
+					std::experimental::default_accessor<ElementType_x>
+				      > x,
+				      std::experimental::mdspan<ElementType_y,
+					std::experimental::extents<ext_y>,
+					Layout_y,
+					std::experimental::default_accessor<ElementType_y>
+				      > y)
+{
+
+  // constraints
+  static_assert(A.rank() == 2);
+  static_assert(x.rank() == 1);
+  static_assert(y.rank() == 1);
+
+  // preconditions
+  if ( A.extent(0) != A.extent(1) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(0) != A.extent(1) ");
+  }
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: hermitian_matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+
+  // for non-complex ELementType_A, A can be treated as symmetric
+  // http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1673r6.html#hermitian-matrix-vector-product-linalgalgsblas2hemv
+  //
+  // If inout_matrix_t::element_type is complex<RA> for some RA, then ....
+  // Otherwise, the functions will assume that A[j,i] equals A[i,j].
+  //
+  if constexpr (hermitianmatvecimpl::is_complex_v<ElementType_A> == false){
+    std::experimental::linalg::symmetric_matrix_vector_product(kexe, A, tr, x, y);
+  }
+  else
+    {
+
+      auto A_view = Impl::mdspan_to_view(A);
+      auto x_view = Impl::mdspan_to_view(x);
+      auto y_view = Impl::mdspan_to_view(y);
+
+      auto ex = ExeSpace();
+
+      if constexpr (std::is_same_v<Triangle, std::experimental::linalg::upper_triangle_t>)
+      {
+
+	// this print is detected in the tests
+#if defined KOKKOS_STDBLAS_ENABLE_TESTS
+	std::cout << "overwriting_hermitian_matrix_vector_product_upper: kokkos impl\n";
+#endif
+
+	Kokkos::parallel_for(Kokkos::RangePolicy(ex, 0, A_view.extent(0)),
+			     KOKKOS_LAMBDA (const std::size_t & i)
```

Comment:
Please see my comments elsewhere on preferring to pass types of size less than or equal to `sizeof(void*)` by value, instead of by const reference; thanks!